### DISCUSSION TITLE
posts/Go/array/

### DISCUSSION THREAD

**Discussion by giscus** [Q1mi/BlogComments#369] [Announcements]
# posts/Go/array/

李文周的Blog

https://www.liwenzhou.com/posts/Go/array/

**Comment by ligongxin** [comment]
func arraySum() {
	//求数组[1, 3, 5, 7, 8]所有元素的和
	a := [5]int{1, 3, 5, 7, 8}
	var count int
	for _, v := range a {
		count += v
	}
	fmt.Println(count)

}

func sumIndex() {
	//找出数组中和为指定值的两个元素的下标，比如从数组[1, 3, 5, 7, 8]中找出和为8的两个元素的下标分别为(0,3)和(1,2)
	a := [5]int{1, 3, 5, 7, 8}
	for i := 0; i < len(a)-1; i++ {
		for j := i+1; j < len(a); j++ {
			if a[i]+a[j] == 8 && i!=j {
				fmt.Println(i, j)
			}
		}
	}
}

**Comment by Lilymz** [comment]
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var array, sum = [...]int{2, 2, 2, 2, 2}, 0
	for _, num := range array {
		sum += num
	}
	fmt.Println("数组总和:" + strconv.Itoa(sum))
	for _, value := range sumIndex(array, 4) {
		fmt.Println("当前为8的元素总和下标:" + value)
	}
}

// 获取指定和元素的下标
func sumIndex(array [5]int, sum int) []string {
	var result []string
	cursor := len(array)
	for i := 0; i < cursor-1; i++ {
		for j := i + 1; j < len(array); j++ {
			if array[i]+array[j] == sum {
				result = append(result, strconv.Itoa(i)+","+strconv.Itoa(j))
			}
		}
	}
	return result
}

**Comment by flyfei00** [comment]
b := [...]int{1, 3, 5, 7, 8}
	for i := 0; i < len(b); i++ {
		for j := i + 1; j < len(b); j++ {
			if b[i]+b[j] == 8 {
				fmt.Printf("%d  %d is Corrent", i, j)
				fmt.Println()
			}
		}
	}

**Comment by Johnnylov** [comment]
package main
import "fmt"
func main() {
	arr := [5]int{1, 3, 5, 7, 8}
	sum := 0
	for index, value := range arr {
		sum += value
		for i := index + 1; i < len(arr); i++ {
			if value+arr[i] == 8 {
				fmt.Printf("(%d,%d)\n", index, i)
			}
		}
	}
	fmt.Println("sum=", sum)
}
